src/saltext/vmware/utils/esxi.py

```diff
@@ -155,3 +159,164 @@ def list_hosts(service_instance):
         The Service Instance Object from which to obtain hosts.
     """
     return utils_common.list_objects(service_instance, vim.HostSystem)
+
+
+def disconnect_host(name, service_instance):
+    """
+    Disconnects host from vCenter instance
+
+    Returns connection state of host
+
+    name
+        Name of host.
+
+    service_instance
+        The Service Instance Object from which to obtain host.
+    """
+    host = utils_common.get_mor_by_property(service_instance, vim.HostSystem, name)
+    if host.summary.runtime.connectionState == "disconnected":
+        return host.summary.runtime.connectionState
+    task = host.DisconnectHost_Task()
+    host = utils_common.wait_for_task(task, name, "disconnect host task")
+    return host.summary.runtime.connectionState
+
+
+def reconnect_host(name, service_instance):
+    """
+    Reconnects host from vCenter instance
+
+    Returns connection state of host
+
+    name
+        Name of host.
+
+    service_instance
+        The Service Instance Object from which to obtain host.
+    """
+    host = utils_common.get_mor_by_property(service_instance, vim.HostSystem, name)
+    if host.summary.runtime.connectionState == "connected":
+        return host.summary.runtime.connectionState
+    task = host.ReconnectHost_Task()
+    ret_host = utils_common.wait_for_task(task, name, "reconnect host task")
+    return ret_host.summary.runtime.connectionState
+
+
+def move_host(name, cluster_name, service_instance):
+    """
+    Move host to a different cluster.
+
+    Returns connection state of host
+
+    name
+        Name of host.
+
+    cluster_name
+        Name of cluster to move host to.
+
+    service_instance
+        The Service Instance Object from which to obtain host.
+    """
+    host_ref = utils_common.get_mor_by_property(service_instance, vim.HostSystem, name)
+    cluster_ref = utils_common.get_mor_by_property(
+        service_instance, vim.ClusterComputeResource, cluster_name
+    )
+    if cluster_ref.parent.parent != host_ref.parent.parent.parent:
+        raise salt.exceptions.VMwareApiError("Cluster has to be in the same datacenter")
+    task = cluster_ref.MoveInto_Task([host_ref])
+    utils_common.wait_for_task(task, cluster_name, "move host task")
+    return "moved"
```

Comment:
**suggestion (non-blocking)** Would it be better to say where it moved from and where it moved to? e.g. `return f"moved {name} from {host_ref.parent.or.whatever} to {cluster_ref}"`?

Also, that raises a question - will it always be the `.parent.parent` and `.parent.parent.parent` of these things that is the datacenter, or is it possible that more or less levels of nesting could happen? Would it be better to do some kind of search up the tree for the datacenter?